### PR TITLE
feat(community): Added Google Scholar Integration

### PR DESCRIPTION
**Pull request by xnnxnyx**
Description:
This PR is to integrate Google Scholar functionality into the LangChain JS version, similar to the existing implementation in Python. This will provide tools for querying Google Scholar and handling the results efficiently within the JavaScript ecosystem.

Issue # [7078](https://github.com/langchain-ai/langchainjs/discussions/7078)

Sample Output: 
```
Search Results: [
  {
    "title": "Machine learning algorithms-a review",
    "link": "https://www.researchgate.net/profile/Batta-Mahesh/publication/344717762_Machine_Learning_Algorithms_-A_Review/links/5f8b2365299bf1b53e2d243a/Machine-Learning-Algorithms-A-Review.pdf?eid=5082902844932096t",
    "snippet": "… Here‟sa quick look at some of the commonly used algorithms in machine learning (ML) Supervised Learning Supervised learning is the machine learning task of learning a function …",
    "publication_info": "International Journal of Science and Research (IJSR) …, 2020 - researchgate.net",
    "authors": "",
    "total_citations": 3511
  },
  {
    "title": "Machine learning",
    "link": "https://books.google.com/books?hl=en&lr=&id=2nQJEAAAQBAJ&oi=fnd&pg=PR7&dq=Machine+Learning&ots=fI04V6ZGlo&sig=2yQM3B3GFmiBmM1u03EFKfTsQ7I",
    "snippet": "MIT presents a concise primer on machine learning—computer programs that learn from data and the basis of applications like voice recognition and driverless cars. No in-depth …",
    "publication_info": "2021 - books.google.com",
    "authors": "E Alpaydin",
    "total_citations": 1155
  },
  {
    "title": "Machine learning",
    "link": "https://books.google.com/books?hl=en&lr=&id=ctM-EAAAQBAJ&oi=fnd&pg=PR6&dq=Machine+Learning&ots=o_LjZ4Wz0o&sig=xnCjg2k2aj1DVlo87V2Q6iUXS0w",
    "snippet": "… machine learning. The second part includes Chapters 4–10, which presents some classic and popular machine learning … cover the core topics of machine learning in one semester, and …",
    "publication_info": "2021 - books.google.com",
    "authors": "ZH Zhou",
    "total_citations": 2734
  },
  {
    "title": "Machine learning: Trends, perspectives, and prospects",
    "link": "https://www.science.org/doi/abs/10.1126/science.aaa8415",
    "snippet": "… Machine learning addresses the question of how to build computers that improve … Recent progress in machine learning has been driven both by the development of new learning …",
    "publication_info": "Science, 2015 - science.org",
    "authors": "MI Jordan, TM Mitchell",
    "total_citations": 10150
  }
]
```

### CODE REVIEW

File: docs/core_docs/docs/integrations/tools/google_scholar.mdx
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/tools/google_scholar.mdx:1]: Can you please use the template here?

https://github.com/langchain-ai/langchainjs/tree/main/libs/langchain-scripts/src/cli/docs/templates

File: examples/src/tools/google_scholar.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { GoogleScholarAPI } from "@langchain/community/tools/google_scholar";
+
+const scholar = new GoogleScholarAPI({
+  apiKey: process.env.SERPAPI_API_KEY
+});
+
+(async () => {
+  try {
+    const query = "Attention is all we need";
+    const results = await scholar._call(query);
```
Comment (jacoblee93) [inline examples/src/tools/google_scholar.ts:10]: Use `.invoke`

File: libs/langchain-community/src/tools/google_scholar.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  /**
+   * Optional API key for accessing the SerpApi service.
+   */
+  apiKey?: string;
+}
+
+/**
+ * Tool for querying Google Scholar using the SerpApi service.
+ */
+export class GoogleScholarAPI extends Tool {
```
Comment (jacoblee93) [inline libs/langchain-community/src/tools/google_scholar.ts:18]: Let's call this `SERPGoogleScholarAPITool`

**Review by jacoblee93 — COMMENTED** [review]
You will also need to add an entrypoint:

https://github.com/langchain-ai/langchainjs/blob/main/.github/contributing/INTEGRATIONS.md

**Comment by xnnxnyx** [comment]
Hi @jacoblee93 thank you for reviewing the PR! I have made the requested changes based on your feedback. When you have a moment, could you please take another look and let me know if there's anything else you'd like adjusted?

**Review by jacoblee93 — APPROVED** [review]
Thanks for your patience!

**Pull request merged**